### PR TITLE
feat: add onclick prop to link component

### PR DESCRIPTION
**Pull request by kunalnagar-okta**
[OKTA-659173](https://oktainc.atlassian.net/browse/OKTA-659173)

## Summary

- Add `component` and `onClick` props to the `Link` component
- Slack context [here](https://okta.slack.com/archives/C7T2H3KNJ/p1697655748363219?thread_ts=1697581182.634639&cid=C7T2H3KNJ)

## Testing & Screenshots

- [x] I have confirmed this change with my designer and the Odyssey Design Team.

Storybook example of the `component` prop in action

image

### CODE REVIEW

File: packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store
Comment (kunalnagar-okta) [inline packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store:1]: I have no idea what this file is doing - looks like it was already in the repo lol?

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store:1]: Ah, this should be deleted. Must've been added in one of the previous PRs.

```sh
git rm --cached /packages/odyssey-react-mui/src/properties/ts/.ts.DS_Store
```

File: packages/odyssey-react-mui/src/Link.tsx [hunk 1]
```diff
@@ -27,10 +27,18 @@ export type LinkProps = {
...
+   */
+  component?: ElementType;
   /**
    * An optional Icon component at the start of the Link
    */
   icon?: ReactElement;
+  /**
+   * The click event handler for the Link
+   */
+  onClick?: MuiLinkProps["onClick"];
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Link.tsx:41]: 👍

File: packages/odyssey-react-mui/src/Link.tsx [hunk 2]
```diff
@@ -58,13 +66,15 @@ const Link = ({
...
   variant,
+  ...rest
 }: LinkProps) => (
   <MuiLink
     data-se={testId}
     href={href}
     rel={rel}
     target={target}
     variant={variant}
+    {...rest}
```
[the remark this answers is not in the document]
Comment (kunalnagar-okta) [reply packages/odyssey-react-mui/src/Link.tsx:77]: The `...rest` is for the Odyssey `LinkProps`, not the `MuiLinkProps`

**Review by KevinGhadyani-Okta — CHANGES_REQUESTED** [review]
I thought we were only adding the `onClick` prop for this PR.

Adding the `component` prop opens a can of worms and warrants a separate discussion. To me, it's tech debt, and that's strictly against the guidelines of Odyssey.

**Review by KevinGhadyani-Okta — APPROVED** [review]

**Pull request merged**